software/spmd/load_dependency/main.S

```diff
@@ -9,30 +9,30 @@ li ra, 10;
 
 sum:
     // Store data in some locations
-    bsg_asm_remote_store(0, 0, 0x1000, 1);
-    bsg_asm_remote_store(0, 0, 0x1004, 2);
-    bsg_asm_remote_store(0, 0, 0x1008, 3);
-    bsg_asm_remote_store(0, 0, 0x100c, 4);
-    bsg_asm_remote_store(0, 0, 0x1010, 5);
-    bsg_asm_remote_store(0, 0, 0x1014, 6);
-    bsg_asm_remote_store(0, 0, 0x1018, 7);
-    bsg_asm_remote_store(0, 0, 0x101c, 8);
-    bsg_asm_local_store(0x1020, 9);
-    bsg_asm_local_store(0x1024, 10);
-    bsg_asm_local_store(0x1028, 11);
-    bsg_asm_local_store(0x102c, 12);
-    bsg_asm_local_store(0x1030, 13);
-    bsg_asm_local_store(0x1034, 14);
-    bsg_asm_local_store(0x1038, 15);
+    bsg_asm_remote_store(0, 0, 0x0000, 1);
+    bsg_asm_remote_store(0, 0, 0x0004, 2);
+    bsg_asm_remote_store(0, 0, 0x0008, 3);
+    bsg_asm_remote_store(0, 0, 0x000c, 4);
+    bsg_asm_remote_store(0, 0, 0x0010, 5);
+    bsg_asm_remote_store(0, 0, 0x0014, 6);
+    bsg_asm_remote_store(0, 0, 0x0018, 7);
+    bsg_asm_remote_store(0, 0, 0x001c, 8);
+    bsg_asm_local_store(0x0020, 9);
+    bsg_asm_local_store(0x0024, 10);
+    bsg_asm_local_store(0x0028, 11);
+    bsg_asm_local_store(0x002c, 12);
+    bsg_asm_local_store(0x0030, 13);
+    bsg_asm_local_store(0x0034, 14);
+    bsg_asm_local_store(0x0038, 15);
     fence;
 
 test_1:
     // Single cycle remote loads
     // nops to test zero delay remote loads
-    bsg_asm_remote_load(s0, 0, 0, 0x1000); // 1
-    bsg_asm_remote_load(s1, 0, 0, 0x1004); // 2
-    bsg_asm_remote_load(s2, 0, 0, 0x1008); // 3
-    bsg_asm_remote_load(s3, 0, 0, 0x100c); // 4
+    bsg_asm_remote_load(s0, 0, 0, 0x0000); // 1
+    bsg_asm_remote_load(s1, 0, 0, 0x0004); // 2
+    bsg_asm_remote_load(s2, 0, 0, 0x0008); // 3
+    bsg_asm_remote_load(s3, 0, 0, 0x000c); // 4
```

Comment:
Is there a symbol we can dereference instead of hardcoding 0x0000?

Reply:
The programmer  knows that DMEM starts from 0.